src/components/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,314 @@
+import type { FC, ReactNode } from 'react';
+import { useEffect, useRef, useState } from 'react';
+import { HiArrowLeft, HiArrowRight, HiCalendar } from 'react-icons/hi';
+import { twMerge } from 'tailwind-merge';
+import type { DeepPartial } from '..';
+import { TextInput, type FlowbiteTextInputTheme, type TextInputProps } from '..';
+import { useTheme } from '../..';
+import { mergeDeep } from '../../helpers/merge-deep';
+import { DatepickerContext } from './DatepickerContext';
+import type { FlowbiteDatepickerViewsDaysTheme } from './Views/Days';
+import { DatepickerViewsDays } from './Views/Days';
+import { DatepickerViewsDecades, type FlowbiteDatepickerViewsDecadesTheme } from './Views/Decades';
+import { DatepickerViewsMonth, type FlowbiteDatepickerViewsMonthsTheme } from './Views/Months';
+import { DatepickerViewsYears, type FlowbiteDatepickerViewsYearsTheme } from './Views/Years';
+import {
+  Views,
+  WeekStart,
+  addMonths,
+  addYears,
+  getFirstDateInRange,
+  getFormattedDate,
+  isDateEqual,
+  startOfYearPeriod,
+} from './helpers';
+
+export interface FlowbiteDatepickerTheme {
+  root: {
+    base: string;
+    input?: FlowbiteTextInputTheme;
+  };
+  popup: FlowbiteDatepickerPopupTheme;
+  views: {
+    days: FlowbiteDatepickerViewsDaysTheme;
+    months: FlowbiteDatepickerViewsMonthsTheme;
+    years: FlowbiteDatepickerViewsYearsTheme;
+    decades: FlowbiteDatepickerViewsDecadesTheme;
+  };
+}
+
+export interface FlowbiteDatepickerPopupTheme {
+  root: {
+    base: string;
+    inline: string;
+    inner: string;
+  };
+  header: {
+    base: string;
+    title: string;
+    selectors: {
+      base: string;
+      button: {
+        base: string;
+        prev: string;
+        next: string;
+        view: string;
+      };
+    };
+  };
+  view: {
+    base: string;
+  };
+  footer: {
+    base: string;
+    button: {
+      base: string;
+      today: string;
+      clear: string;
+    };
+  };
+}
+
+export interface DatepickerProps extends Omit<TextInputProps, 'theme'> {
```

Comment:
Missing a prop here for the user to receive the selected date, like an `onChange(date: Date)` or `onSelect(date: Date)`

Reply:
This is kinda crucial!

Reply:
Yes please!